### PR TITLE
PT-2441: Candidate genes should not list an information button on patient view.

### PR DESCRIPTION
**Pull request by veronikaslc**
Fixed.

### CODE REVIEW

File: components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__genes.xml [hunk 1]
```diff
@@ -137,10 +137,14 @@
         var infoTool = new Element('span', {'class' : 'fa fa-info-circle xHelpButton gene-info', 'title' : geneNames[i]});
         var icon_helpController = new PhenoTips.widgets.HelpButton(infoTool);
         var triangle = new Element('span', {'class': "fa fa-exclamation-triangle", title: '$services.localization.render('phenotips.patientSheetCode.termSuggest.nonStandardGenesymbol')'}).update(' ');
-        if (isSuggestSelected)
-          element.insert({after: infoTool});
-        else
+        if (isSuggestSelected) {
+          if (!$('inline')) {
```
Comment (marta-) [inline components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__genes.xml]: Just FYI, there's `XWiki.contextaction`  (which can be `view`, `edit`, `inline`, `admin`) available in the JS API. No need the modify the code though, this is also fine.

Comment (marta-) [inline components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__genes.xml]: But shouldn't the condition be the other way around? If we're in edit mode, only then insert the info button?

Comment (veronikaslc) [inline components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__genes.xml]: Sure, stupit typo :)

**Comment by veronikaslc** [comment]
@itaiGershtansky @mjshepherd please, review

File: components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__genes.xml [hunk 2]
```diff
@@ -137,10 +137,14 @@
         var infoTool = new Element('span', {'class' : 'fa fa-info-circle xHelpButton gene-info', 'title' : geneNames[i]});
         var icon_helpController = new PhenoTips.widgets.HelpButton(infoTool);
         var triangle = new Element('span', {'class': "fa fa-exclamation-triangle", title: '$services.localization.render('phenotips.patientSheetCode.termSuggest.nonStandardGenesymbol')'}).update(' ');
-        if (isSuggestSelected)
-          element.insert({after: infoTool});
-        else
+        if (isSuggestSelected) {
+          if ($('inline')) {
```
Comment (mjshepherd) [inline components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__genes.xml]: this could be one if statement.

Comment (mjshepherd) [inline components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__genes.xml]: Also the infoTool and triangle elements are not conditionally generated even though they are conditionally placed on the screen.

Comment (mjshepherd) [inline components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__genes.xml]: Also it isn't clear from reading the code what ```if ($('inline'))``` is actually checking for. I guess its currently checking for the existence of an element which is only present in edit mode? But I can only take that guess by looking at the issue title.

Comment (veronikaslc) [inline components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__genes.xml]: Indeed, that helped me to find much better fix.

**Comment by veronikaslc** [comment]
@mjshepherd updated and rebased

File: components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__genes.xml [hunk 3]
```diff
@@ -137,10 +137,12 @@
         var infoTool = new Element('span', {'class' : 'fa fa-info-circle xHelpButton gene-info', 'title' : geneNames[i]});
         var icon_helpController = new PhenoTips.widgets.HelpButton(infoTool);
         var triangle = new Element('span', {'class': "fa fa-exclamation-triangle", title: '$services.localization.render('phenotips.patientSheetCode.termSuggest.nonStandardGenesymbol')'}).update(' ');
```
Comment (mjshepherd) [inline components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__genes.xml]: I'm still not a fan that triangle and infotool are generated regardless of whether they are used.

File: components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__genes.xml [hunk 4]
```diff
@@ -137,10 +137,12 @@
...
-        if (isSuggestSelected)
+        if (isSuggestSelected) {
```
Comment (mjshepherd) [inline components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__genes.xml]: I'm still not a fan that triangle and infotool are generated regardless of whether they are used.

Comment (veronikaslc) [inline components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__genes.xml]: Agreed. This is exactly why `fetchGeneDetails()` function that generates them is now called only `if (XWiki.contextaction == "edit")`, [lines 171-173].

Comment (mjshepherd) [inline components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__genes.xml]: yes thats good but its not what I mean. Also is the variable ```isSuggestSelected``` still appropriately named. Would something like ```isViewMode``` be better?

Comment (veronikaslc) [inline components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__genes.xml]: The name pretty much reflects the context. If suggested term is selected then the info icon is inserted. If not - the triangle icon. It has nothing to do with view mode. I put brackets there just for code improvement, this is not the actual bugfix. The actual bugfix happens in lines 171-173 where I restrict the whole function that generates those icons to be implemented only in view mode.

**Comment by veronikaslc** [comment]
@itaiGershtansky @mjshepherd : updared PR with codereview comments.

**Comment by mjshepherd** [comment]
+1

**Comment by marta-** [comment]
It was decided to not make this change, for usability reasons.

**Pull request closed without merging**